src/java.base/share/classes/java/text/DecimalFormat.java

```diff
@@ -134,120 +154,89 @@
- * negative subpattern. That is, {@code "0.00"} alone is equivalent to
- * {@code "0.00;-0.00"}.  If there is an explicit negative subpattern, it
- * serves only to specify the negative prefix and suffix; the number of digits,
- * minimal digits, and other characteristics are all the same as the positive
- * pattern. That means that {@code "#,##0.0#;(#)"} produces precisely
- * the same behavior as {@code "#,##0.0#;(#,##0.0#)"}.
- *
- * <p>The prefixes, suffixes, and various symbols used for infinity, digits,
- * grouping separators, decimal separators, etc. may be set to arbitrary
- * values, and they will appear properly during formatting.  However, care must
- * be taken that the symbols and strings do not conflict, or parsing will be
- * unreliable.  For example, either the positive and negative prefixes or the
- * suffixes must be distinct for {@code DecimalFormat.parse()} to be able
- * to distinguish positive from negative values.  (If they are identical, then
- * {@code DecimalFormat} will behave as if no negative subpattern was
- * specified.)  Another example is that the decimal separator and grouping
- * separator should be distinct characters, or parsing will be impossible.
- *
- * <p>The grouping separator is commonly used for thousands, but in some
- * countries it separates ten-thousands. The grouping size is a constant number
- * of digits between the grouping characters, such as 3 for 100,000,000 or 4 for
- * 1,0000,0000.  If you supply a pattern with multiple grouping characters, the
- * interval between the last one and the end of the integer is the one that is
- * used. So {@code "#,##,###,####"} == {@code "######,####"} ==
- * {@code "##,####,####"}.
- *
  * <h3><a id="special_pattern_character">Special Pattern Characters</a></h3>
  *
- * <p>Many characters in a pattern are taken literally; they are matched during
- * parsing and output unchanged during formatting.  Special characters, on the
- * other hand, stand for other characters, strings, or classes of characters.
+ * <p>The special characters in the table below are interpreted syntatically when
+ * used in the DecimalFormat pattern.
  * They must be quoted, unless noted otherwise, if they are to appear in the
  * prefix or suffix as literals.
  *
- * <p>The characters listed here are used in non-localized patterns.  Localized
- * patterns use the corresponding characters taken from this formatter's
- * {@code DecimalFormatSymbols} object instead, and these characters lose
- * their special status.  Two exceptions are the currency sign and quote, which
- * are not localized.
+ * <p>When calling {@link #applyPattern(String)}, use the default symbols in the
+ * table. When calling {@link #applyLocalizedPattern(String)} use the corresponding
+ * localized symbol.
+ * When {@link #applyLocalizedPattern(String)} is called, the default symbols lose their
+ * syntactical meaning, and vice versa with {@link #applyPattern(String)} with exception
```

Comment:
Retain the original wording that explains localized and non-localized patterns at the beginning. Using the method example for the localized patterns might be a bit jump in the context.